src/ActionList/Item.tsx

```diff
@@ -251,8 +305,39 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
                   </Box>
                   {slots.inlineDescription}
                 </ConditionalBox>
-                {slots.trailingVisual}
+                {
```

Comment:
I was able to clean up this logic. See https://github.com/primer/react/pull/3913/files#r1409960232